packages/sirius-web/frontend/sirius-web-application/src/views/project-browser/list-projects-area/ListProjectsArea.tsx

```diff
@@ -34,37 +33,165 @@ const useListProjectsAreaStyles = makeStyles()((theme) => ({
+        afterRefreshProjects: false,
+        beforeRefreshNeeded: false,
+        afterRefreshNeeded: false,
+        previousPageRequired: false,
+        nextPageRequired: false,
+      }));
+    } else if (state.nextPageRequired && !afterLoading && afterData) {
+      setState((prevState) => ({
+        ...prevState,
+        projects: afterData?.viewer.projects.edges.map((edge) => edge.node) ?? [],
+        hasPrev: true,
+        hasNext: afterData?.viewer.projects.pageInfo.hasNextPage ?? false,
+        startCursor: afterData?.viewer.projects.pageInfo.startCursor,
+        endCursor: afterData?.viewer.projects.pageInfo.endCursor,
+        onRefreshProjects: afterRefreshProjects,
+        beforeRefreshProjects: false,
+        afterRefreshProjects: true,
+        beforeRefreshNeeded: false,
+        afterRefreshNeeded: false,
+        previousPageRequired: false,
+        nextPageRequired: false,
+      }));
+    } else if (state.beforeRefreshNeeded && !beforeLoading && beforeData) {
+      setState((prevState) => ({
+        ...prevState,
+        projects: beforeData?.viewer.projects.edges.map((edge) => edge.node).reverse() ?? [],
+        hasPrev: beforeData?.viewer.projects.pageInfo.hasPreviousPage ?? false,
+        hasNext: true,
+        startCursor: beforeData?.viewer.projects.pageInfo.endCursor,
+        endCursor: beforeData?.viewer.projects.pageInfo.startCursor,
+        onRefreshProjects: beforeRefreshProjects,
+        beforeRefreshProjects: true,
+        afterRefreshProjects: false,
+        beforeRefreshNeeded: false,
+        afterRefreshNeeded: false,
+        previousPageRequired: false,
+        nextPageRequired: false,
+      }));
+    } else if (state.afterRefreshNeeded && !afterLoading && afterData) {
+      setState((prevState) => ({
+        ...prevState,
+        projects: afterData?.viewer.projects.edges.map((edge) => edge.node) ?? [],
+        hasPrev: true,
+        hasNext: afterData?.viewer.projects.pageInfo.hasNextPage ?? false,
+        startCursor: afterData?.viewer.projects.pageInfo.startCursor,
+        endCursor: afterData?.viewer.projects.pageInfo.endCursor,
+        onRefreshProjects: afterRefreshProjects,
+        beforeRefreshProjects: false,
+        afterRefreshProjects: true,
+        beforeRefreshNeeded: false,
+        afterRefreshNeeded: false,
+        previousPageRequired: false,
+        nextPageRequired: false,
+      }));
+    }
+  }, [afterLoading, beforeLoading, afterData, beforeData, afterRefreshProjects, beforeRefreshProjects]);
+
+  const onPreviousPage = () => {
+    setState((prevState) => ({ ...prevState, previousPageRequired: true }));
+  };
+
+  const onNextPage = () => {
+    setState((prevState) => ({ ...prevState, nextPageRequired: true }));
+  };
+
+  const onPageSizeChange = (pageSize: number) =>
+    setState((prevState) => ({ ...prevState, pageSize, startCursor: null, endCursor: null }));
+
+  const onRefreshProjects = () => {
+    state.onRefreshProjects();
```

Comment:
HELP NEEDED : I don't know how to handle the refresh of a project (i.e. rename or delete).
`state.onRefreshProjects();` launches a refetch of data, but I don't know how to properly handle `hasPrev/hasNext` after this refresh.
The `setState` statements below are not a good idea, because they fetch `useProjectsBefore` and `useProjectsAfter` in addition to the `refetch`.